marge/job.py

```diff
@@ -167,6 +172,7 @@ def wait_for_ci_to_pass(self, merge_request, commit_sha=None):
 
         log.info('Waiting for CI to pass for MR !%s', merge_request.iid)
         while datetime.utcnow() - time_0 < self._options.ci_timeout:
+            pipeline = self.get_pipeline_by_mr(merge_request, commit_sha=commit_sha)
             ci_status = self.get_mr_ci_status(merge_request, commit_sha=commit_sha)
```

Comment:
Since get_mr_ci_status is I believe only used here, probably better to just make this `self.get_pipeline_ci_status(pipeline)` to avoid extra requests and potential inconsistencies between the pipeline fetched at t and t+1?